api/services/metadata_service.py

```diff
@@ -154,6 +153,7 @@ def enable_built_in_field(dataset: Dataset):
                     document.doc_metadata = doc_metadata
                     db.session.add(document)
             db.session.commit()
+            dataset.built_in_field_enabled = True
```

Comment:
Moving this line down will not update the dataset.built_in_field_enabled in the database because the session is already commited.